Add main script to run `integrative-analysis` module (#76)
stjude-dnb-binfcore/sc-rna-seq-snap: analyses/integrative-analysis/01-integrative-analysis.Rmd

---
title: "Integrative analysis for sc-/sn-RNA-Seq Analysis in 10X Genomics data"
author: "Antonia Chroni for SJCRH DNB_BINF_Core"

> author: "Antonia Chroni for SJCRH DNB_BINF_Core"

Reviewer: author line is duplicated

Reviewer: ```suggestion
```

Author: Resolved here b5c1ce8017ebfec75b0df9dc348a7a2922970556

Hence, integration of single-cell sequencing datasets, for example across experimental batches, donors, or conditions, is often an important step in scRNA-seq workflows. Integrative analysis can help to match shared cell types and states across datasets, which can boost statistical power, and most importantly, facilitate accurate comparative analysis across datasets. Here, we include the following integration methods for integrative analysis:

(1) [Seurat ‘anchor-based’](https://satijalab.org/seurat/articles/seurat5_integration) integration workflow by [Choudhary & Satija, 2022](https://doi.org/10.1186/s13059-021-02584-9),
(2) [Harmony](https://portals.broadinstitute.org/harmony/) integration workflow by [Korsunsky et al., 2019](https://www.nature.com/articles/s41592-019-0619-0), and
(3) [Liger](https://welch-lab.github.io/liger/articles/liger_with_seurat.html) integration workflow by [Welch et al., 2019](https://doi.org/10.1016/j.cell.2019.05.006).

> Different number of dimensions for different integration algorithms (Seurat, Harmony, Liger, etc.) is common and depending on the algorithm they come with their corresponding variables depending on how the algorithm is being used. Each of these integration algorithm takes into account typically the same variables but the math they use to transform them is so different that you can't use the same number of resolutions. Even though they use the same word, resolution doesn't actually effect each algorithm the same way and their maximums can be different. 30 for Seurat, 30 for Harmony and 20 for Liger give the most similar integration results when all other variables are taken into account while also not wasting too much compute time.

Reviewer: The first sentence is a bit hard to follow. I tried to make a suggestion to clarify it, but I'm not sure what will sound best.

Also, are dimension and resolution being used interchangeably here? If so, perhaps it is better to stick with using dimension throughout for clarity. And to not cause confusion with the resolution parameter which is used for clustering.

```suggestion
Using a different number of dimensions for different integration algorithms (Seurat, Harmony, Liger, etc.) is common and their corresponding variables depend on the algorithm and how the algorithm is being used. Each of these integration algorithms takes into account typically the same variables but the math they use to transform them is so different that you can't use the same number of resolutions. Even though they use the same word, resolution doesn't actually effect each algorithm the same way and their maximums can be different. 30 for Seurat, 30 for Harmony and 20 for Liger give the most similar integration results when all other variables are taken into account while also not wasting too much compute time.
```
